### ISSUE TITLE
[A11y] ClearIndicator violation 

### ISSUE THREAD

**Issue by Kretajak** [JedWatson/react-select#4793]
When using custom ClearIndicator along with isClearable set to true, after selecting one item - AXE is reporting `ARIA hidden element must not contain focusable elements`.

*Steps to reproduce:*

- Open: https://798fh.csb.app/
- Select any item from dropdown
- Clear button should be visible
- Now run `axe`. 

*Result:*

Axe is reporting `ARIA hidden element must not contain focusable elements` for custom Clear Indicator. 

*Workaround:* 

I can set `aria-hidden="true"` to fix the axe issue.

*Comment:* 

This is because as a clear indicator i'm using `<button>` which is focusable by tab. However, this button is getting `aria-hidden="true"` always. No matter it's present or not. I understand this forced `true` is coming from https://github.com/JedWatson/react-select/pull/2581/files. However what's the point of hiding this useful clear indicator from screen readers? It can be extremely useful for clearing multiple options at once but not only in this case.

**Comment by ebonow** [comment]
Greetings @Kretajak ,

We've recently been taking a much closer look at accessibility and this was something that has been discussed. 

The rationale for this decision is so that the clear indicator does not interfere with the keyboard navigation by adding an extra tab to leave the input. While this may seem strange, it is seems to be consistent with the native input type="search" where a clear indicator/button appears but is not "tabbable".

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search

If you would like to override the default `aria-hidden="true"` you can either remove the `aria-hidden` attribute or override its value to false in the attributes. Note that there might be a better way to destructure variable names with hyphens so that this could be done within the scope of destructing innerProps so that the spread operator doesn't include it in restInnerProps.

``` jsx
const ClearIndicator = (props) => {
  const {
    innerProps: { ref, ...restInnerProps }
  } = props;

  delete restInnerProps["aria-hidden"];

  return (
    <button {...restInnerProps} ref={ref}>
      Clear all
    </button>
  );
};
```

https://codesandbox.io/s/react-select-a11y-issue-forked-ufbsr?file=/src/index.js

Hope this is helpful and please let us know if you would like to discuss more about the clearIndicator being more accessible and we can take it up for discussion.

**Comment by Kretajak** [comment]
Thanks @ebonow for quick response. I see your point. 

In my case overriding `aria-hidden` to `true` is all I need to solve this violation. I will move forward with it.
